src/iterresult.ts

```diff
@@ -40,58 +39,53 @@ export default class IterResult<M extends QueryMethodTypes> {
   }
 
   /**
-   * Possibly adds a date into the result.
+   * Checks a date against the boundary conditions
    *
-   * @param {Date} date - the date isn't necessarly added to the result
-   *                      list (if it is too late/too early)
-   * @return {Boolean} true if it makes sense to continue the iteration
-   *                   false if we're done.
+   * @param {Date} date
+   * @return {Boolean} true this is a date of interest
    */
   accept (date: Date) {
     ++this.total
     const tooEarly = this.minDate && date < this.minDate
     const tooLate = this.maxDate && date > this.maxDate
 
     if (this.method === 'between') {
-      if (tooEarly) return true
+      if (tooEarly) return false
       if (tooLate) return false
     } else if (this.method === 'before') {
       if (tooLate) return false
     } else if (this.method === 'after') {
-      if (tooEarly) return true
-      this.add(date)
-      return false
+      if (tooEarly) return false
+      return true
     }
 
-    return this.add(date)
+    return true
   }
```

Comment:
Accept was split into two parts, accept and shouldContinue. Prior to this the function was implementing both pieces of logic in the same function.